pkg/kube/inject/inject.go

```diff
@@ -1066,38 +1087,12 @@ func rewriteCniPodSpec(annotations map[string]string, spec *SidecarInjectionSpec
 	}
 }
 
-// overwriteClusterInfo updates cluster name and network from url path
-// This is needed when webconfig config runs on a different cluster than webhook
-func overwriteClusterInfo(containers []corev1.Container, path string) {
+// splitPath gets the clsuter name and network from the injection URL path
+func splitPath(path string) (clusterName string, clusterNetwork string) {
 	res := strings.Split(path, "/")
 	if len(res) >= 5 {
-		// if len is less than 5, not enough length for /cluster/X/net/Y
-		clusterName, clusterNetwork := "", ""
-		clusterName = res[len(res)-3]
-		clusterNetwork = res[len(res)-1]
-
-		log.Debugf("Updating cluster info based on clusterName: %s clusterNetwork: %s\n", clusterName, clusterNetwork)
-
-		for i, c := range containers {
-			if c.Name == ProxyContainerName {
-				updateClusterInfo(&containers[i], clusterName, clusterNetwork)
-			}
-		}
+		return res[len(res)-3], res[len(res)-1]
 	}
-
-}
-
-func updateClusterInfo(container *corev1.Container, clusterName, clusterNetwork string) {
-	envVars := make([]corev1.EnvVar, 0)
-	for _, env := range container.Env {
-		if env.Name != "ISTIO_META_CLUSTER_ID" && env.Name != "ISTIO_META_NETWORK" {
-			envVars = append(envVars, env)
-		}
-	}
-
-	log.Debugf("Appending env ISTIO_META_CLUSTER_ID: %s and ISTIO_META_NETWORK: %s\n", clusterName, clusterNetwork)
-	envVars = append(envVars,
-		corev1.EnvVar{Name: "ISTIO_META_CLUSTER_ID", Value: clusterName, ValueFrom: nil},
-		corev1.EnvVar{Name: "ISTIO_META_NETWORK", Value: clusterNetwork, ValueFrom: nil})
-	container.Env = envVars
+	// if len is less than 5, not enough length for /cluster/X/net/Y
+	return "", ""
```

Comment:
It seems you are removing code - any indication that code is not used by someone ? Can we maintain backward compat and minimize upgrade risks here ?

Reply:
This code was used to set env vars on the containers after the templates were executed. I just moved that into the templates.

Reply:
@stevenctl it's a little tough to review since it wasn't just a simple move. It looks like the new method only handles network, not cluster. I'm concerned that there is missing logic now.

Reply:
`updateClusterInfo` goes through and patches the injection spec with the cluster/network that we get from the URL. This grabs the values from the URL earlier so we can just set the right values on the first pass.

If we don't do this, anything that would set cluster/network that isn't part of `values` needs to do this more complex patching.

Reply:
@linsun I think you may have set this up as part of central istiod, can you confirm this move would work ok?